Modules/posixmodule.c

```diff
@@ -9577,38 +9577,49 @@ wait_helper(PyObject *module, pid_t pid, int status, struct rusage *ru)
     if (!result)
         return NULL;
 
+    PyObject *pidObject = PyLong_FromPid(pid);
+    if (pidObject == NULL) {
+        goto error;
+    }
```

Comment:
I think that this change is not actually needed, because `Py_BuildValue()` handles NULL argument with exception set pretty well. `Py_BuildValue("NiN", PyLong_FromPid(pid), status, result)` is correct.